src/machi_file_proxy.erl

```diff
@@ -355,24 +342,25 @@ handle_call({append, ClientMeta, Extra, Data}, _From,
     ClientCsumTag = proplists:get_value(client_csum_tag, ClientMeta, ?CSUM_TAG_NONE),
     ClientCsum = proplists:get_value(client_csum, ClientMeta, <<>>),
 
-    {Resp, NewErr, NewU} =
+    {Resp, NewErr} =
     case check_or_make_tagged_csum(ClientCsumTag, ClientCsum, Data) of
         {error, {bad_csum, Bad}} ->
             lager:error("Bad checksum; client sent ~p, we computed ~p",
                         [ClientCsum, Bad]),
-            {{error, bad_checksum}, Err + 1, U};
+            {{error, bad_checksum}, Err + 1};
         TaggedCsum ->
-            case handle_write(FHd, CsumTable, F, TaggedCsum, EofP, Data, U) of
-                {ok, NewU1} ->
-                    {{ok, F, EofP}, Err, NewU1};
+            case handle_write(FHd, CsumTable, F, TaggedCsum, EofP, Data) of
+                ok ->
+                    {{ok, F, EofP}, Err};
                 Error ->
-                    {Error, Err + 1, EofP, U}
+                    {Error, Err + 1, EofP}
             end
     end,
-    {NewEof, infinity} = lists:last(NewU),
+    %% TODO: do we check this with calling
+    %% machi_csum_table:calc_unwritten_bytes/1?
+    NewEof = EofP + byte_size(Data) + Extra,
```

Comment:
I think it's ok to handle this directly in the file proxy code - it's pretty clear what's going on here and the file proxy is still the "sequencer" that decides the offsets for an append, so it makes sense to keep that in the file proxy state IMO.